### PR TITLE
[#736] Add highlighting test cases to the Xtext Homeautomation example.

### PR DESCRIPTION
**Pull request by miklossy**
- Extend the Xtext Homeautomation example by the
RuleEngineHighlightingTest to demonstrate the usage of the
AbstractHighlightingTest abstract test class how to assert on
highlighting configuration text styles.
- Add corresponding entries to the build.xml and to the plugin.xml to
ensure that the homeautomation.ui.tests plugin is zipped and packaged.

Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

### CODE REVIEW

**Comment by kthoms** [comment]
Thx, @miklossy

File: org.eclipse.xtext.xtext.ui.examples/projects/homeautomation/org.eclipse.xtext.example.homeautomation.ui.tests/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Bundle-ActivationPolicy: lazy
+Require-Bundle: org.eclipse.xtext.example.homeautomation.ui,
+ org.eclipse.xtext.testing,
+ org.eclipse.core.runtime,
+ org.eclipse.ui.workbench;resolution:=optional,
+ org.eclipse.xtext.common.types.ui,
+ org.eclipse.jdt.core,
+ org.eclipse.xtext.xbase.testing,
+ org.eclipse.xtext.ui.testing,
+ org.eclipse.xtext.junit4,
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/homeautomation/org.eclipse.xtext.example.homeautomation.ui.tests/META-INF/MANIFEST.MF]: do we still need this or is xtext.testing supposed to be sufficient?

File: org.eclipse.xtext.xtext.ui.examples/projects/homeautomation/org.eclipse.xtext.example.homeautomation.ui.tests/src/org/eclipse/xtext/example/homeautomation/ui/tests/RuleEngineHighlightingTest.xtend [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+@RunWith(XtextRunner)
+@InjectWith(RuleEngineUiInjectorProvider)
+class RuleEngineHighlightingTest extends AbstractHighlightingTest {
+
+	@Inject extension DefaultHighlightingConfiguration
+
+	@Test def device_keyword() {
+		'''
+			Device Window can be open, closed
+		'''.testHighlighting("Device", keywordTextStyle)
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/homeautomation/org.eclipse.xtext.example.homeautomation.ui.tests/src/org/eclipse/xtext/example/homeautomation/ui/tests/RuleEngineHighlightingTest.xtend]: I like this a lot :)

File: org.eclipse.xtext.xtext.ui.examples/projects/homeautomation/org.eclipse.xtext.example.homeautomation.ui.tests/META-INF/MANIFEST.MF (hunk 1 above)
Comment (miklossy) [reply org.eclipse.xtext.xtext.ui.examples/projects/homeautomation/org.eclipse.xtext.example.homeautomation.ui.tests/META-INF/MANIFEST.MF]: Thanks for the hint! I will investigate that and (wherever possible) clean up the Require-Bundle sections of all the examples in a follow up PR.

**Pull request merged**